### ISSUE TITLE
How get key when push object?

### ISSUE THREAD

**Issue by danielfnz** [adriancarriger/angularfire2-offline#16]
How get key when push object?

   test.push({
...
    })
     .then((item) => this._successCadastro(item))
      .catch((err) => this._errorCadastro(err));
  }
  private _successCadastro(item) {
    console.log(item.$key);
  }

undefined variable

**Comment by danielfnz** [comment]
@adriancarriger

**Comment by mediastuttgart** [comment]
+1

**Comment by adriancarriger** [comment]
Thanks for reporting this! I can confirm that this is a bug and the promise should return the related key in a fashion [similar to AngularFire2](https://github.com/angular/angularfire2/issues/199#issuecomment-245074714).

**Comment by adriancarriger** [comment]
This should now be fixed in version 2.0.5

To get a key on push try this:

```ts
const promise = this.groceries.push({text: 'Milk'});
console.log(promise.key);
```

You do not need to use then or wait for the promise to complete.
